routing/blinding.go

```diff
@@ -0,0 +1,69 @@
+package routing
+
+import (
+	"errors"
+	"fmt"
+
+	sphinx "github.com/lightningnetwork/lightning-onion"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// ErrNoBlindedPath is returned when the blinded path in a blinded
+	// payment is missing.
+	ErrNoBlindedPath = errors.New("blinded path required")
+
+	// ErrInsufficientBlindedHops is returned when a blinded path does
+	// not have enough blinded hops.
+	ErrInsufficientBlindedHops = errors.New("blinded path requires " +
+		"at least one hop")
+)
+
+// BlindedPayment provides the path and payment parameters required to send a
+// payment along a blinded path.
+type BlindedPayment struct {
```

Comment:
Also, a spec note: there's still some ongoing discussion around the [specifics relating to bolt12 invoices](https://github.com/lightning/bolts/pull/1069). Doesn't matter to us since we're doing it in reverse, but wanted to flag!